From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Wed, 1 May 2024 10:57:48 +0200
Subject: layout-manager: Use min padding with zero radius
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

A radius of zero means that there's no value present in the data.
This affect devices that have a panel information but no border radius
specified (like e.g. the Librem 5).

Fixes: 474b07f90 ("layout-manager: Allow to query rounded corner information")

Signed-off-by: Guido Günther <agx@sigxcpu.org>
Part-of: <https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1424>
(cherry picked from commit 2e335ff4cab62b40f69e26d9c83eae3fee28db00)
---
 src/layout-manager.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/layout-manager.c b/src/layout-manager.c
index 8ccbd83..16086d8 100644
--- a/src/layout-manager.c
+++ b/src/layout-manager.c
@@ -176,6 +176,9 @@ get_corner_shift (PhoshLayoutManager *self)
   scale = phosh_monitor_get_fractional_scale (self->builtin);
   r = c = gm_display_panel_get_border_radius (self->panel) / scale;
 
+  if (G_APPROX_VALUE (r, 0, FLT_EPSILON))
+    goto out;
+
   /*
    * We want to keep the n pixels towards the top screen edge clear
    * n + b = r and m + a = r and c = r
